### PR TITLE
[feature](array_function) add support for array_popfront

### PR DESCRIPTION
**Pull request by bobhan1**
# Proposed changes

Issue Number: #17361 

`array_popfront()` remove the first element from array.
usage example:
```
mysql> set enable_vectorized_engine=true;

mysql> select array_popfront(['test', NULL, 'value']);
+-----------------------------------------------------+
| array_popfront(ARRAY('test', NULL, 'value'))        |
+-----------------------------------------------------+
| [NULL, value]                                       |
+-----------------------------------------------------+

mysql> select array_popfront([]);
+---------------------------+
| array_popfront(ARRAY())   |
+---------------------------+
| []                        |
+---------------------------+
```
## Problem summary

Describe your changes.

## Checklist(Required)

* [x] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
@Yukang-Lian

**Review by Yukang-Lian — COMMENTED** [review]
Please add regression tests and docs.

**Comment by bobhan1** [comment]
> Please add regression tests and docs.

Done

**Comment by Yukang-Lian** [comment]
> > Please add regression tests and docs.
> 
> Done

We need add test results to this PR, plz follow this guide to generate `.out` file.

https://doris.incubator.apache.org/zh-CN/community/developer-guide/regression-testing#%E4%BD%BF%E7%94%A8%E6%9F%A5%E8%AF%A2%E7%BB%93%E6%9E%9C%E8%87%AA%E5%8A%A8%E7%94%9F%E6%88%90out%E6%96%87%E4%BB%B6

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy [hunk 1]
```diff
@@ -99,6 +99,12 @@ suite("test_array_functions") {
     qt_select "SELECT k1, array_popback(k7) from ${tableName} ORDER BY k1"
     qt_select "SELECT k1, array_popback(k8) from ${tableName} ORDER BY k1"
     qt_select "SELECT k1, array_popback(k10) from ${tableName} ORDER BY k1"
+    qt_select "SELECT k1, array_popfront(k2) from ${tableName} ORDER BY k1"
```
Comment (eldenmoon) [inline regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:102]: please add null cases

Comment (eldenmoon) [reply regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions.groovy:102]: test_array_functions_by_literal.groovy you can easily write cases by literals

File: docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_popfront.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## array_popfront
+
+<version since="1.2.0">
```
Comment (eldenmoon) [inline docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_popfront.md:27]: version 1.2.3 or 2.0

**Review by Yukang-Lian — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions_by_literal.groovy [hunk 3]
```diff
@@ -189,6 +189,18 @@ suite("test_array_functions_by_literal") {
...
+    qt_sql "select array_popfront([])"
+    qt_sql "select array_popfront(null)"
+    qt_sql "select array_popfront([null,2,3,4,5])"
+    qt_sql "select array_popfront([1,2,3,4,null])"
+    qt_sql "select array_popfront(['1','2','3','4','5','6'])"
+    qt_sql "select array_popfront([null,'2','3','4','5','6'])"
+    qt_sql "select array_popfront(['1','2','3','4','5',null])"
+    qt_sql "select array_popfront(array(cast ('2023-02-06' as datev2), cast ('2023-02-05' as datev2), cast ('2023-02-07' as datev2), cast ('2023-02-05' as datev2)))"
+    qt_sql "select array_popfront(array(null, cast ('2023-02-06' as datev2), cast ('2023-02-05' as datev2), cast ('2023-02-07' as datev2), cast ('2023-02-05' as datev2)))"
+
```
Comment (Yukang-Lian) [inline regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions_by_literal.groovy:203]: please add a test for `datetimev2`.

**Comment by eldenmoon** [comment]
seems p0 regression failed

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by bobhan1** [comment]
这个FE UT build failure从日志好像看不出什么，是还需要再重跑一遍吗

**Comment by Yukang-Lian** [comment]
run feut

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
regression test似乎又因为一些其他原因挂了?

**Comment by bobhan1** [comment]
regression test似乎又因为一些其他原因挂了?

> regression test似乎又因为一些其他原因挂了?

我rebase一下再推试试

**Comment by bobhan1** [comment]
run p0

**Review by Yukang-Lian — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**